extensions/mssql/src/util/dataCache.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export class DataItemCache<T> {
+
+	millisecondsToLive: number;
+	getValueFunction: (...args: any[]) => Promise<T>;
```

Comment:
Does this have to be any[]? Is there a way we can use generics for this as well?

Reply:
I was trying to think of a way but there could be any variable number of arguments and they can be of any type combinations. Putting in something like `(...args: E[]) => Promise<T>;` would require all args to be of the same type. This proposal [Give Specific Types to Variadic Functions](https://github.com/microsoft/TypeScript/issues/5453) looks interesting.